agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4121,32 +4102,49 @@ bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMess
         if (!autoconfig_wsc_authenticate(m2, authkey))
             return false;
 
-        bool fronthaul, backhaul, teardown;
-        auto credentials = request->create_wifi_credentials();
-        if (!autoconfig_wsc_parse_m2_encrypted_settings(m2, authkey, keywrapkey, backhaul,
-                                                        fronthaul, teardown, credentials))
-            return false;
-
-        LOG(DEBUG) << "Controller configuration (WSC M2 Encrypted Settings)";
-        LOG(DEBUG) << "     Manufacturer: " << m2.manufacturer()
-                   << ", ssid: " << credentials->ssid_str()
-                   << ", bssid: " << credentials->bssid_attr().data
-                   << ", authentication_type: " << std::hex
-                   << int(credentials->authentication_type_attr().data)
-                   << ", encryption_type: " << int(credentials->encryption_type_attr().data)
-                   << (fronthaul ? " fronthaul" : "") << (backhaul ? " backhaul" : "")
-                   << (teardown ? " teardown" : "");
+        WSC::configData::config config_;
+        if (!autoconfig_wsc_parse_m2_encrypted_settings(m2, authkey, keywrapkey, config_)) {
+            LOG(ERROR) << "Invalid config data, tear down radio";
```

Comment:
I think we'll need to revisit this at some point. From a security perspective, if authentication fails, we should do nothing instead of tearing down. If config data is invalid it's a bit uncertain what we should do.

Reply:
You are absolutely right - I rather do this right now than postpone it.
I changed the way we parse config data so that there is a difference between encryption/authentication failures - in which we ignore the config data, and empty config_data which is also invalid according to Table 20 in the WSC specification but Marvell controller sends it none the less.